src/server_manager/web_app/ui_components/outline-per-key-data-limit-dialog.ts

```diff
@@ -0,0 +1,380 @@
+        }
+
+        paper-listbox paper-item:hover {
+          cursor: pointer;
+          background-color: var(--background-contrast-color);
+          color: #fff;
+        }
+      `,
+    ];
+  }
+
+  /**
+   * @member _keyName The displayed name of the UI access key representing the key we're working on.
+   */
+  @internalProperty() _keyName = '';
+  /** @member _keyId The id of the UI access key representing the key we're working on. */
+  @internalProperty() _keyId = '';
+  /**
+   * @member _keyDataLimitBytes The data limit, if it exists, on the access key we're working on.
+   */
+  @internalProperty() _keyDataLimitBytes: number = undefined;
+  /**
+   * @member serverDefaultLimitBytes The default data limit of the server we're working on, or null
+   */
+  @internalProperty() _serverDefaultLimitBytes: number = undefined;
+  /**
+   * @member _showMenu Whether the menu for inputting the data limit should be shown. Controlled by
+   * the checkbox.
+   */
+  @internalProperty() _showMenu = false;
+  /**
+   * @member _enableSave Whether the save button is enabled.  Controlled by the validator on the
+   * input.
+   */
+  @internalProperty() _enableSave = false;
+  /**
+   * @member _language The language used for i18n
+   */
+  @internalProperty() _language = 'en';
+  @property({type: Function}) localize: Function;
+
+  private _serverId = '';
+
+  render() {
+    return html`
+      <style>
+        /* Custom element mixins with brackets don't work in style() */
+        #unitsDropdown {
+          --paper-input-container-underline: {
+            display: none;
+          }
+          --paper-input-container-underline-focus: {
+            display: none;
+          }
+        }
+        #dataLimitInput {
+          --paper-input-container-label-focus: {
+            color: rgb(123, 123, 123);
+          }
+        }
+      </style>
+      <paper-dialog id="container">
+        <div id="headerSection">
+          <iron-icon id="dataLimitIcon" icon="icons:perm-data-setting"></iron-icon>
+          <h3>${this.localize('per-key-data-limit-dialog-title', 'keyName', this._keyName)}</h3>
+        </div>
+        <div id="menuSection">
+          <paper-checkbox ?checked=${this._showMenu} @tap=${this._setCustomLimitTapped}>
+            ${this.localize('per-key-data-limit-dialog-set-custom')}
+          </paper-checkbox>
+          ${this._showMenu ? this.renderMenu() : ''}
+        </div>
+        <div id="buttonsSection">
+          <paper-button id="save" ?disabled=${!this._enableSave} @tap=${this._sendSaveEvent}>${
+        this.localize('save')}</paper-button>
+          <paper-button @tap=${this.close}>${this.localize('cancel')}</paper-button>
+        </div>
+      </paper-dialog>
+    `;
+  }
+
+  private renderMenu() {
+    return html`
+      <div id="menu">
+        <paper-input
+          id="dataLimitInput"
+          label=${this.localize('data-limit')}
+          always-float-label
+          allowed-pattern="[0-9\\.]"
```

Comment:
I wonder how well this internationalized.
Different countries use different number separators.

Reply:
A demonstration of the problem here:

```js
// Easy Case
Number.parseFloat("1.23")
1.23
// First issue
Number.parseFloat("1,23")
1
// Second issue
Number.parseFloat("١٫٢٣")
NaN
```

The first issue we could solve pretty easily. The second issue is a problem I've wrestled with and punted on.  I think it's *possible* to build a localized number parser based on `Intl.NumberFormat.formatToParts`, but that seems like a problem for a different PR, and anyways we haven't heard of any issues with users having trouble with the default data limit input.  We could match the default data limit input and only allow integers, although I'd like in that case to make the same restriction on the server so there isn't skew between the display and the server state.

It would be nice for `Intl` to have localized parsing to go with its localized formatting.